rotkehlchen/db/updates.py

```diff
@@ -327,6 +332,32 @@ def update_global_addressbook(self, data: list[dict[str, Any]], version: int) ->
                 entries=entries_to_add,
             )
 
+    def update_location_asset_mappings(self, data: dict[str, list[dict[str, Any]]], version: int) -> None:  # noqa: E501
+        """Applies location asset mappings updates in the global DB"""
+        log.info(f'Applying update for location asset mappings to v{version}')
+        for update_function, entry_type, raw_data_key in (
+            (GlobalDBHandler.add_location_asset_mappings, LocationAssetMappingUpdateEntry, 'additions'),  # noqa: E501
+            (GlobalDBHandler.update_location_asset_mappings, LocationAssetMappingUpdateEntry, 'updates'),  # noqa: E501
+            (GlobalDBHandler.delete_location_asset_mappings, LocationAssetMappingDeleteEntry, 'deletions'),  # noqa: E501
+        ):
+            entries, raw_data = [], data.get(raw_data_key)
+            if raw_data is None:
+                continue  # no data to update
+            for raw_entry in raw_data:
+                try:
+                    if 'asset' in raw_entry:
+                        raw_entry['asset'] = Asset(raw_entry['asset'])
+                    if raw_entry['location'] is not None:
+                        raw_entry['location'] = Location.deserialize(raw_entry['location'])
```

Comment:
Do this only if location can be missing. If not then ignore

```suggestion
                    if (raw_location := raw_entry.get('location')) is not None:
                        raw_entry['location'] = Location.deserialize(raw_location)
```